application/apps/webassembly/matcher/src/lib.rs

```diff
@@ -25,81 +26,81 @@ impl Matcher {
     pub fn new() -> Self {
         Self {
             matcher: SkimMatcherV2::default(),
-            items: String::new(),
+            items_initial: Vec::new(),
+            items_scored: HashMap::new(),
         }
     }
 
     #[wasm_bindgen]
-    pub fn set_items(&mut self, items: String) {
-        self.items = items;
-    }
-
-    #[wasm_bindgen]
-    pub fn search_single(
-        &self,
-        query: String,
-        item: String,
-        tag: Option<String>,
-    ) -> Result<String, String> {
-        if query.is_empty() {
-            return Ok(item);
-        }
-        match self.matcher.fuzzy_indices(&item, &query) {
-            Some(score) => match self.tag_match(item, score.1, &tag) {
-                Ok(tagged_match) => Ok(tagged_match),
-                Err(err) => Err(err),
-            },
-            None => Ok(item),
+    pub fn set_item(&mut self, item: String) -> Result<usize, String> {
+        match serde_json::from_str(&item) {
+            Ok::<HashMap<String, String>, _>(item) => {
+                self.items_initial.push(item);
+                match self.search(String::new(), None) {
+                    Ok(_) => {}
+                    Err(err) => {
+                        return Err(format!("Initializing with empty search failed: {}", err))
+                    }
+                }
```

Comment:
```
match self.search(String::new(), None) {
                    Ok(_) => {}
                    Err(err) => {
                        return Err(format!("Initializing with empty search failed: {}", err))
                    }
                }
```

                
this block can be replaced with: `self.search(String::new(), None).map_err(|e| format!("Initializing with empty search failed: {}", e))?;`